src/pages/ORCID_redirect.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * A ORCID redirect page
+ */
+
+import Cookies from 'universal-cookie';
+import {
+  useEffect,
+  useState
+} from "react";
+import { useNavigate } from "react-router-dom";
+import 'css/ORCIDRedirect.css';
+
+function ORCIDPage() {
+  const [token, setToken] = useState({});
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const code = window.location.hash.split("?")[1];
+    const param = new URLSearchParams(code);
+
+    // fetch ORCID fields from iSamples endpoint
+    fetch(`${window.config.ordic_token}?code=${param.get('code')}`)
+      .then((token) => token.json())
+      .then((res) => {
+        setToken(res);
+        const cookies = new Cookies();
+        const expired = res['expires_in'];
+        if (res === 'Failure') { return };
+        cookies.set('access_token', res['access_token'], { path: '/', expires: new Date(Date.now() + expired) });
+        cookies.set('refresh_token', res['refresh_token'], { path: '/', expires: new Date(Date.now() + expired) });
+        cookies.set('orcid', res['orcid'], { path: '/' });
+        cookies.set('name', res['name'], { path: '/' });
```

Comment:
This part might have some bugs. Although the endpoint has already set up ORCID cookies, the returned pages cannot be detected by our react app because of the hash router. On this redirect page, I need to fetch the result from the endpoint and set the cookies.

Reply:
Yeah, that's the confusing part. Since I used isamples_inabox and isamples_webui in separate servers, I still have a problem running isamples_docker.